src/service/autoUploadService.ts

```diff
@@ -0,0 +1,82 @@
+import { watch } from "chokidar";
+import * as vscode from "vscode";
+import Commons from "../commons";
+import { Environment } from "../environmentPath";
+import lockfile from "../lockfile";
+import { CustomSettings } from "../setting";
+import { FileService } from "./fileService";
+
+export class AutoUploadService {
+  public static GetIgnoredItems(customSettings: CustomSettings) {
+    return [
+      ...customSettings.ignoreUploadFolders.map(folder => `**/${folder}/**`),
+      ...customSettings.ignoreUploadFiles.map(file => `**/${file}`)
+    ];
+  }
+
+  public watching = false;
+
+  private watcher = watch(this.options.en.USER_FOLDER, {
+    depth: 2,
+    ignored: this.options.ignored
+  });
+
+  constructor(
+    private options: { en: Environment; commons: Commons; ignored: string[] }
+  ) {
+    vscode.extensions.onDidChange(async () => {
+      if (this.watching) {
```

Comment:
Thanks for adding. :)
Does new extension adds on `extensions.json` while on upload process?  Previously we needed to restart the code like before.

Reply:
After I installed an extension, it auto-uploaded and showed up in my gist.